clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -350,8 +351,12 @@ private void pollSelectionKeys(Iterable<SelectionKey> selectionKeys,
                 }
 
                 /* if channel is not ready finish prepare */
-                if (channel.isConnected() && !channel.ready())
+                if (channel.isConnected() && !channel.ready()) {
                     channel.prepare();
+                    if (channel.ready()) {
+                        this.sensors.connectionPrepared.record();
```

Comment:
@edoardocomar `Prepared` doesn't convey much meaning in terms of an externally visible metric. I imagine you chose it rather than `authenticated` since you intended it to work for `PLAINTEXT`. But `PLAINTEXT` doesn't go through this if-block since `channel.ready()` returns `true`.